lib/collection/src/recommendations.rs

```diff
@@ -26,44 +26,80 @@ use crate::operations::types::{
     RecommendRequestInternal, RecommendStrategy, UsingVector,
 };
 
-// TODO(sparse): support avg for sparse vectors
-fn avg_vectors<'a>(vectors: impl Iterator<Item = VectorRef<'a>>) -> Vector {
-    let mut count: usize = 0;
-    let mut avg_vector: VectorType = Default::default();
-    for vector in vectors {
-        match vector {
-            VectorRef::Dense(vector) => {
+// TODO(sparse): what if vectors iterator is empty? We added CollectionError::BadRequest,
+// but it's not clear if it's the best solution
+fn avg_vectors<'a>(mut vectors: impl Iterator<Item = VectorRef<'a>>) -> CollectionResult<Vector> {
```

Comment:
can we instead partition the `vectors` using match dense/sparse to see if the iterator is homogeneous?

Reply:
fixed